app/view_models/prior_authority/service_costs_with_adjustments.rb

```diff
@@ -0,0 +1,139 @@
+
+    def requested_humanized_cost_per_unit
+      i18n_key = cost_type == 'per_item' ? "per_#{item_type}" : cost_type
+
+      "#{requested_base_cost_per_unit} " \
+        "#{I18n.t(i18n_key, scope: 'prior_authority.application_details.items.per_unit_descriptions')}"
+    end
+
+    def requested_formatted_service_cost_total
+      NumberTo.pounds(requested_base_cost)
+    end
+
+    private
+
+    def requested_items
+      original_items
+    end
+
+    def requested_period
+      original_period
+    end
+
+    def requested_base_cost
+      if cost_type == 'per_item'
+        requested_items * requested_cost_per_item
+      else
+        (
+          (requested_period.hours * requested_cost_per_hour) +
+          ((requested_period.minutes / 60.0) * requested_cost_per_hour)
+        ).round(2)
+      end
+    end
+
+    def requested_base_cost_per_unit
+      NumberTo.pounds(requested_cost_per_unit)
+    end
+
+    def requested_cost_per_unit
+      @requested_cost_per_unit ||= cost_type == 'per_item' ? requested_cost_per_item : requested_cost_per_hour
+    end
+
+    def requested_cost_per_item
+      original_cost_per_item
+    end
+
+    def requested_cost_per_hour
+      original_cost_per_hour
+    end
+  end
+
+  module AdjustedServiceCosts
+    include RequestedServiceCosts
+
+    def adjusted_humanized_units
+      return unless adjusted_items || adjusted_period
+
+      if cost_type == 'per_item'
+        "#{adjusted_items} " \
+          "#{I18n.t("prior_authority.application_details.items.#{item_type}").pluralize(adjusted_items)}"
+      else
+        format_period(adjusted_period)
+      end
+    end
+
+    def adjusted_humanized_cost_per_unit
+      return unless adjusted_formatted_cost_per_unit
+
+      i18n_key = cost_type == 'per_item' ? "per_#{item_type}" : cost_type
+
+      "#{adjusted_formatted_cost_per_unit} " \
+        "#{I18n.t(i18n_key, scope: 'prior_authority.application_details.items.per_unit_descriptions')}"
+    end
+
+    def adjusted_formatted_service_cost_total
+      NumberTo.pounds(adjusted_service_cost_total) if adjusted_service_cost_total
+    end
+
+    private
+
+    def adjusted_formatted_cost_per_unit
+      NumberTo.pounds(adjusted_cost_per_unit) if adjusted_cost_per_unit
+    end
+
+    def adjusted_cost_per_unit
+      @adjusted_cost_per_unit ||= cost_type == 'per_item' ? adjusted_cost_per_item : adjusted_cost_per_hour
+    end
+
+    def adjusted_service_cost_total
+      cost_type == 'per_item' ? adjusted_item_cost_total : adjusted_hour_cost_total
+    end
+
+    def adjusted_item_cost_total
+      return unless adjusted_items && adjusted_cost_per_item
+
+      adjusted_items * adjusted_cost_per_item
+    end
+
+    def adjusted_hour_cost_total
+      return unless adjusted_period && adjusted_cost_per_hour
+
+      (
+        (adjusted_period.hours * adjusted_cost_per_hour) +
+        ((adjusted_period.minutes / 60.0) * adjusted_cost_per_hour)
+      ).round(2)
+    end
+
+    def adjusted_items
```

Comment:
The logic of these next few methods is a bit unintuitive to me. Because "adjusted_items" will return a value if "items" hasn't been adjusted (if cost_per_item _has_ been adjusted). This method is kinda trying to do too much. I think it would make more sense to me to separate out a method called `any_per_item_adjustments?` that would check for `items_original` and `cost_per_item_original`. The places that currently call `adjusted_items` in a guard clause could instead call `any_per_item_adjustment?`. And the other places that currently call `adjusted_items` could then just call `items` instead.

Reply:
refactored as suggested and was able to remove the `adjusted_..` pseudo attribute wrappers, so more succinct and hopefully simpler/clearer.